src/components/Delegate/StakingOpportunities.tsx

```diff
@@ -73,10 +74,83 @@ export const ValidatorName = ({ moniker, isStaking }: ValidatorNameProps) => {
   )
 }
 
-export const StakingOpportunities = ({ assetId, opportunities }: StakingOpportunitiesProps) => {
-  const isLoaded = true
-  const isStaking = opportunities.some(x => x.cryptoAmount)
-  const assetSymbol = useAppSelector(state => selectAssetByCAIP19(state, assetId)).symbol
+export const StakingOpportunities = ({ assetId }: StakingOpportunitiesProps) => {
+  const stakingDataStatus = useAppSelector(selectStakingDataStatus)
+  const isStakingDataLoaded = stakingDataStatus === 'loaded'
+  const validatorStatus = useAppSelector(selectValidatorStatus)
+  const isValidatorDataLoaded = validatorStatus === 'loaded'
+  const isLoaded = isStakingDataLoaded && isValidatorDataLoaded
+  const asset = useAppSelector(state => selectAssetByCAIP19(state, assetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+  const dispatch = useAppDispatch()
+
+  const accountSpecifiersForChainId = useAppSelector(state =>
+    selectPubkeyishByChainId(state, asset?.caip2)
+  )
+  const accountSpecifier = accountSpecifiersForChainId?.[0]
+
+  const activeStakingOpportunities = useAppSelector(state =>
+    selectActiveStakingOpportunityDataByDenom(
+      state,
+      accountSpecifier,
+      SHAPESHIFT_VALIDATOR_ADDRESS,
+      ASSET_ID_TO_DENOM[asset.caip19]
+    )
+  )
+
+  const shapeshiftValidator = useAppSelector(state =>
+    selectSingleValidator(state, accountSpecifier, SHAPESHIFT_VALIDATOR_ADDRESS)
+  )
+  const stakingOpportunityDefault = [
+    {
+      ...shapeshiftValidator
+    }
+  ]
+  const nonLoadedValidators = useAppSelector(state =>
+    selectNonloadedValidators(state, accountSpecifier)
+  )
+  const isStaking = activeStakingOpportunities.length !== 0
+
+  useEffect(() => {
+    ;(async () => {
+      if (!accountSpecifier?.length || isStakingDataLoaded) return
+
+      dispatch(
+        stakingDataApi.endpoints.getStakingData.initiate(
+          { accountSpecifier },
+          { forceRefetch: true }
+        )
+      )
+    })()
+  }, [accountSpecifier, isStakingDataLoaded, dispatch])
+
+  useEffect(() => {
+    ;(async () => {
+      if (isValidatorDataLoaded) return
+
+      dispatch(
+        stakingDataApi.endpoints.getAllValidatorsData.initiate(
+          { chainId: 'cosmos:cosmoshub-4' },
```

Comment:
Can we not hardcode the chainId? We have access to `assetId` in this component, so we can do:
```
const { chain } = caip19.fromCAIP19(assetId)
```

And then in the query, use `byChain` instead of `byChainId`